### PR TITLE
:seedling: update golangci-lint action

### LINKED ISSUE #2878: Update golangci-lint to latest in GitHub Actions
The golangci-lint check seems to be failing on multiple PRs with exit code 137 - meaning that the process has run out of memory. 

In an attempt to resolve this, we should bump the golangci-lint action to the latest version.

### PR DESCRIPTION
**Pull request by everettraven**
Signed-off-by: Bryce Palmer <bpalmer@redhat.com>

## Description of change
- Update `golangci/golangci-lint-action` to `v3`
- Update `golangci-lint` version used to `v1.48`
- Add `actions/setup-go@v3` to setup Go for the `golangci/golangci-lint-action` action

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -30,12 +30,16 @@ jobs:
     # Pull requests from the same repository won't trigger this checks as they were already triggered by the push
     if: (github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository) && needs.check_docs_only.outputs.skip != 'true'
     steps:
+      - name: Setup Go
```
Comment (camilamacedo86) [inline .github/workflows/lint.yml:33]: We need to update the Makefile target either.
Otherwise, all fine on this one.

**Review by jakobmoellersap — COMMENTED** [review]
I think we should not say this fixes https://github.com/kubernetes-sigs/kubebuilder/issues/2878 in the issue description until we merged it in and rerun the lint on some of the failing PRs since I think the memory consumption pattern from 1.45 to 1.48 did not change dramatically. Other than that: 🚀 🌔

**Review by camilamacedo86 — APPROVED** [review]
/lgtm
/approved

**Pull request merged**